vertical-pod-autoscaler/recommender/recommender.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/recommender/cluster"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	v1lister "k8s.io/kubernetes/pkg/client/listers/core/v1"
+)
+
+// Recommender recommend resources for certain containers, based on utilization periodically got from metrics api.
+type Recommender interface {
+	Run()
+}
+
+type recommender struct {
+	specClient              cluster.SpecClient
+	metricsFetchingInterval time.Duration
+}
+
+// Currently it just prints out current utilization to the console.
+// It will be soon replaced by something more useful.
+func (r *recommender) RunOnce() {
+	glog.V(3).Infof("Recommender Run")
+
+	podSpecs, err := r.specClient.GetPodSpecs()
+	if err != nil {
+		glog.Errorf("Cannot get SimplePodSpecs. Reason: %+v", err)
+	}
+	for n, spec := range podSpecs {
+		glog.Infof("SimplePodSpec #%v: %+v", n, spec)
+	}
+}
+
+func (r *recommender) Run() {
+	for {
+		select {
+		case <-time.After(time.Second * 30):
+			{
+				r.RunOnce()
+			}
+		}
+	}
+}
+
+// NewRecommender creates a new recommender instance,
+// which can be run in order to provide continuous resource recommendations for containers.
+// It requires cluster configuration object and duration between recommender intervals.
+func NewRecommender(config *rest.Config, metricsFetcherInterval time.Duration) Recommender {
+	recommender := &recommender{
+		specClient:              newSpecClient(config),
+		metricsFetchingInterval: metricsFetcherInterval,
+	}
+	glog.V(3).Infof("New Recommender created %+v", recommender)
+
+	return recommender
+}
+
+func newSpecClient(config *rest.Config) cluster.SpecClient {
+	kubeClient := kube_client.NewForConfigOrDie(config)
+	podLister := newPodLister(kubeClient)
+	return cluster.NewSpecClient(podLister)
+}
+
+// Creates PodLister, listing only not terminated pods.
```

Comment:
I suppose we want to detect terminated pods in order to remove them from the model.

Reply:
So I will maybe change it to:
PodSucceeded + PodFailed + PodRunning
I guess we don't realy need pods in creation (PodPending) or of unknown state (PodUnknown)?

Reply:
I ended up changing it to " NOT pending AND NOT unknown" since OR is not supported in selectors :)